### PR TITLE
Tests for vector_of_arrays.jl

### PR DESCRIPTION
**Pull request by tldrma**

### CODE REVIEW

File: test/vector_of_arrays.jl [hunk 1]
```diff
@@ -97,6 +97,20 @@ using ArraysOfArrays: full_consistency_checks, append_elemptr!
         @test vcat(B1, B2, B3, B4) == vcat(A1, A2, A3, A4)
         @test eltype(vcat(B1, B2, B3, B4)) == Array{Float64,3}
         full_consistency_checks(vcat(B1, B2, B3, B4))
+
+        B1_copy = copy(B1); B3_copy = copy(B3)
+        append!(B1_copy, B3_copy)
```
Comment (oschulz) [inline test/vector_of_arrays.jl]: Maybe add an @inferred here, to check type-stability of `copy`?

File: test/vector_of_arrays.jl [hunk 2]
```diff
@@ -97,6 +97,20 @@ using ArraysOfArrays: full_consistency_checks, append_elemptr!
...
+        @test B1_copy.data == vcat(B1.data, B3.data)
+
+    end
+
+    @testset "indexing" begin
+        V1 = VectorOfArrays(ref_AoA3(Float32, 3))
```
Comment (oschulz) [inline test/vector_of_arrays.jl]: Some `@inferred`s here wouldn't hurt. :-)

File: test/vector_of_arrays.jl [hunk 3]
```diff
@@ -97,8 +97,45 @@ using ArraysOfArrays: full_consistency_checks, append_elemptr!
...
+
+## _view_reshape_spec not yet implemented ##
+#       V1_copy = copy(V1)
+#       V2_copy = copy(V2)
+#       @test setindex!(V1_copy, V1, 1) == V1
+#       setindex!(V2_copy, V2, 1)
+#       @test V2_copy[1] == V2_copy[2]
+
+## function mul(s) not yet implemented ##
+#       sizehint!(v12_copy, 2, (2,2,3))
```
Comment (tldrma) [inline test/vector_of_arrays.jl]: What are these two missing functions supposed to do? Should I try to implement them?

**Comment by oschulz** [comment]
Is this still WIP?

File: test/vector_of_arrays.jl (hunk 3 above)
Comment (oschulz) [reply test/vector_of_arrays.jl]: No, not yet.

**Comment by tldrma** [comment]
Feel free to merge if it's acceptable. I'm not sure how many more tests you want written for this, because maybe other tests would be written differently with those two missing functions added in?

I've just been looking at array_of_similar_arrays.jl and getting a feel for how nested arrays are constructed and organized. I will have a PR started for that tonight/tomorrow.

**Comment by tldrma** [comment]
Ready to be merged.

**Comment by oschulz** [comment]
Thanks!

**Pull request merged**